funcbench/main.go

```diff
@@ -0,0 +1,354 @@
+	if err != nil {
+		return nil, err
+	}
+	_, _, err = execCommand("git", "config", "--global", "user.name", "Prombench Bot Junior")
+	if err != nil {
+		return nil, err
+	}
+
+	branch, err := ioutil.ReadFile("/github/home/commentMonitor/BRANCH")
+	c := gitClient{
+		branch: string(branch),
+		owner:  *event.GetRepo().Owner.Login,
+		repo:   *event.GetRepo().Name,
+	}
+
+	log.Printf("Git Client: %+v\n", c)
+	return &c, err
+}
+
+func (c *gitClient) cloneRepository() error {
+	_, _, err := execCommand("git", "clone", fmt.Sprintf("https://github.com/%s/%s.git", c.owner, c.repo))
+	if err != nil {
+		return err
+	}
+	err = os.Chdir(c.repo)
+	return err
+}
+
+// checkoutToPullRequest applies changes from the pull request to the working tree
+// of the branch that is being compared.
+func (c *gitClient) checkoutPR(num int) error {
+	_, _, err := execCommand("git", "fetch", "origin", fmt.Sprintf("pull/%d/head:pullrequest", num))
+	if err != nil {
+		return err
+	}
+	_, _, err = execCommand("git", "checkout", c.branch)
+	if err != nil {
+		return err
+	}
+	_, exitCode, err := execCommand("git", "merge", "--squash", "--no-commit", "pullrequest")
+	if err != nil || exitCode != 0 {
+		return errors.Wrap(err, "Pull request merge failed.")
+	}
+	_, _, err = execCommand("git", "reset")
+	return err
+}
+
+func (c *gitClient) revertPRChanges() error {
+	return filepath.Walk(os.Getenv("GITHUB_WORKSPACE"), func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.IsDir() || strings.Contains(path, ".git/") || strings.HasSuffix(info.Name(), "test.go") {
+			return nil
+		}
+
+		data, _, err := execCommand("git", "checkout", "--", path)
+		if err != nil {
+			log.Println(data)
+
+			// New file appeared and is not tracked, then it can be removed.
+			if strings.Contains(data, "did not match any file") {
+				data, _, err = execCommand("rm", path)
+				if err != nil {
+					return errors.Errorf("Error: %v; Command out: %s", err, string(data))
+				}
+				return nil
+			}
+			return err
+		}
+		return nil
+	})
+}
+
+func newBenchmarkTester() (*benchmarkTester, error) {
+	if err := os.Setenv("GO111MODULE", "on"); err != nil {
+		log.Fatalln(err)
+	}
+
+	benchRegex, err := ioutil.ReadFile("/github/home/commentMonitor/REGEX")
+	if err != nil {
+		return nil, err
+	}
+	raceArgument, err := ioutil.ReadFile("/github/home/commentMonitor/RACE")
+	if err != nil {
+		return nil, err
+	}
+
+	bench := benchmarkTester{
+		raceFlagEnabled: string(raceArgument) != "-no-race",
+		home:            os.Getenv("HOME"),
+		bechRegex:       string(benchRegex),
+	}
+
+	_, _, err = execCommand("go", "get", "golang.org/x/tools/cmd/benchcmp")
+
+	log.Printf("Benchmark Tester: %+v\n", bench)
+	return &bench, err
+}
+
+func (bench *benchmarkTester) compareBenchmarks(old, new string) (string, error) {
+	out, _, err := execCommand(strings.Join([]string{os.Getenv("GOPATH"), "/bin/benchcmp"}, ""), old, new)
+	log.Println(out)
```

Comment:
please add more context - I mean something like:
```suggestion
	log.Println("benchcmp result:", data)
```

the same for the other log that is missing context.